example/javalib/android/1-hello-world/build.mill

```diff
@@ -0,0 +1,137 @@
+import mill._
+import mill.scalalib._
+import mill.api.PathRef
+import requests._
+import mill.javalib.android.AndroidModule
+
+object app extends AndroidModule {
+
+  val currentFolder = os.Path(millSourcePath.toString.replace("app", ""))
+  val appName = "HelloWorld"
+  
+  def generateResources = T {
+    installAndroidSdk()
+    val genDir = T.dest / "gen"
+    os.makeDir.all(genDir)
+    os.proc(
+      aapt,
+      "package",
+      "-f",
+      "-m",
+      "-J", genDir,
+      "-M", currentFolder / "AndroidManifest.xml",
+      "-I", androidJarPath
+    ).call()
+    PathRef(genDir)
+  }
+
+  def compileJava = T {
+    val objDir = T.dest / "obj"
+    os.makeDir.all(objDir)
+    os.proc(
+      "javac",
+      "-classpath", androidJarPath,
+      "-d", objDir,
+      os.walk(generateResources().path).filter(_.ext == "java"),
+      os.walk(currentFolder / "java").filter(_.ext == "java")
+    ).call()
+    PathRef(objDir)
+  }
+
+  def createJar = T {
+    val jarPath = T.dest / "my_classes.jar"
+    os.proc(
+      d8,
+      os.walk(compileJava().path).filter(_.ext == "class"),  
+      "--output", jarPath,
+      "--no-desugaring"  
+    ).call()
+    PathRef(jarPath)
+  }
```

Comment:
This should not be necessary to do ourselves, we should use the standard `JavaModule` mechanisms to do this

Reply:
we can extend JavaModule but we have to do these tasks like this to insure proper application integration with further steps